core/src/storage2/traits/impls/prims.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::max;
+use crate::{
+    env::{
+        AccountId,
+        Hash,
+    },
+    storage2::traits::{
+        KeyPtr,
+        PackedLayout,
+        SpreadLayout,
+    },
+};
+use ink_prelude::{
+    boxed::Box,
+    string::String,
+};
+use ink_primitives::Key;
+
+macro_rules! impl_layout_for_primitive {
+    ( $($ty:ty),* $(,)? ) => {
+        $(
+            impl_always_packed_layout!($ty, deep: false);
+            impl PackedLayout for $ty {
+                #[inline(always)]
+                fn pull_packed(&mut self, _at: &Key) {}
+                #[inline(always)]
+                fn push_packed(&self, _at: &Key) {}
+                #[inline(always)]
+                fn clear_packed(&self, _at: &Key) {}
+            }
+        )*
+    };
+}
+#[rustfmt::skip]
+impl_layout_for_primitive!(
+    // We do not include `f32` and `f64` since Wasm contracts currently
+    // do not support them. We might add them to this list once we add
+    // support for those primitives.
+    Key, Hash, AccountId,
+    String,
+    bool,
+    u8, u16, u32, u64, u128,
+    i8, i16, i32, i64, i128,
+);
+
+impl<T> SpreadLayout for Option<T>
+where
+    T: SpreadLayout,
+{
+    const FOOTPRINT: u64 = 1 + <T as SpreadLayout>::FOOTPRINT;
+    const REQUIRES_DEEP_CLEAN_UP: bool = <T as SpreadLayout>::REQUIRES_DEEP_CLEAN_UP;
+
+    fn push_spread(&self, ptr: &mut KeyPtr) {
+        match self {
+            Some(value) => {
+                false.push_spread(ptr);
```

Comment:
should be `true`

Reply:
i start to feel that we should generally use `u8` here instead of booleans. It is also more general and can be extended to work on enums in general that way.